File: ProgrammingBasicsJava/Jan2020/LoopsPart2Exercise/src/_04Walking.java
import java.util.Scanner;

public class _04Walking {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        String command = scanner.nextLine();
        int stepsCount = 0;

        while (!command.equals("Going home")) {
            int stepsWalked = Integer.parseInt(command);
            stepsCount += stepsWalked;

            if (stepsCount >= 10000) break;


            command = scanner.nextLine();
        }

        if (command.equals("Going home")) {
            int stepsToHome = Integer.parseInt(scanner.nextLine());
            stepsCount += stepsToHome;
        }

        if (stepsCount >= 10000) {
            System.out.println("Goal reached! Good job!");
        } else {
            stepsCount = 10000 - stepsCount;
            System.out.printf("%d more steps to reach goal.", stepsCount);
        }
    }
}

File: JavaScriptBasics/05_2_WhileLoopExercise/04_Walking.js
function printIsStepsTargetAchieved(input) {
    let index = 0;
    let command = input[index];
    let steps = 0;
    let isTargetReached = false;

    while (command !== "Going home") {
        steps += Number(command);
        index++;

        if (steps >= 10000) {
            console.log(`Goal reached! Good job!\n${steps - 10000} steps over the goal!`);
            isTargetReached = true;
            break;
        }

        command = input[index];
    }

    if (isTargetReached === false) {
        index++;
        command = input[index];
        steps += Number(command);

        if (steps >= 10000) {
            console.log(`Goal reached! Good job!\n${steps - 10000} steps over the goal!`);
        } else {
            console.log(`${10000 - steps} more steps to reach goal.`);
        }
    }
}

printIsStepsTargetAchieved(["1500",
"3000",
"250",
"1548",
"2000",
"Going home",
"2000"]);